sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -3532,6 +3532,49 @@ class DataFrameSuite extends QueryTest
       }.isEmpty)
     }
   }
+
+  test("SPARK-41207: Fix BinaryArithmetic with negative scale") {
+    withSQLConf(SQLConf.LEGACY_ALLOW_NEGATIVE_SCALE_OF_DECIMAL_ENABLED.key -> "true") {
```

Comment:
I don't think we need to add new features for a legacy config, unless this is a regression. Can you clarify which commit caused the regression?